packages/python/src/components/FunctionDeclaration.tsx

```diff
@@ -82,19 +111,12 @@ export interface InitFunctionDeclarationProps
  *
  * @remarks
  *
- * This is a convenience component that sets the name to `__init__`, marks it as
- * an instance function, and forces the name to be `__init__` without applying
- * the name policy.
+ * This is a convenience component for dunder methods.
  */
-export function InitFunctionDeclaration(props: InitFunctionDeclarationProps) {
```

Comment:
`__init__` does function as a regular dunder method (though an ergonomics argument could be made for why we should support it explicitly).

I think we do need to special-case `__new__` though; it's a method that is implicitly a classmethod (i.e. first argument is `cls`) without having a `@classmethod` decorator.

There's also the fact that `__new__` has a fixed call signature, but so do most builtin dunder methods. So unless we want to explicitly define that for all such dunder methods (probably not) I expect it's best to leave it up to the user to create the right call signature.

If it helps, I note that TypeScript doesn't seem to have any special components to handle `constructor()` methods, even though they are "special".